### PR TITLE
fix: optional git config for techdocs feedback

### PR DESCRIPTION
**Pull request by erikxiv**
Signed-off-by: Erik Larsson <erik.larsson@schibsted.com>

Closes https://github.com/backstage/backstage/issues/5438

## Hey, I just made a Pull Request!

If github or gitlab config is missing, techdocs crash on docs that have repo_url/edit_url specified (to provide edit/feedback links) with the following error message: `Uncaught Error: Missing required config value at 'integrations.gitlab'`. This PR makes these config values optional, as in techdocs gracefully handles missing config values.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by OrkoHunter — APPROVED** [review]
Totally

**Comment by kuangp** [comment]
thanks! this should fix #5438

**Comment by OrkoHunter** [comment]
@backstage/maintainers Please approve and merge :)

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts
Comment (Rugvip) [inline plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts:39]: This doesn't always read the value correctly, should be `integration.getString("host")` instead. I know it was that way before, but wanna fix as part of this PR? :D Will get rid of the `any` type too

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.test.ts
Comment (Rugvip) [inline plugins/techdocs/src/reader/transformers/addGitFeedbackLink.test.ts:21]: Just use `new ConfigReader(...)` to mock this instead

**Review by Rugvip — COMMENTED** [review]
👍 , but think we should take the opportunity to clean up the code a bit. Could even move to using the `IntegrationsApi` if you wanna go that far 😁

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts
Comment (erikxiv) [reply plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts]: I gave it a go, stealing with pride from other locations. WDYT?

File: plugins/techdocs/src/plugin.ts
Comment (Rugvip) [inline plugins/techdocs/src/plugin.ts:56]: Let's not, this is supposed to be provided by the app and not any particular plugin

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts
Comment (Rugvip) [inline plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts:43]: should be able to drop the second check now, since reading from the integrations API will always include `github.com`. Same goes for GitLab below

**Review by Rugvip — COMMENTED** [review]
Thanks! 👍

The fewer places with custom integration config reading logic the better 😁

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts
Comment (erikxiv) [reply plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts:43]: According to tests, the edit link should be available even if there is no matching integration config, if the source URL match github/gitlab, as I understood them?

File: plugins/techdocs/src/plugin.ts
Comment (erikxiv) [reply plugins/techdocs/src/plugin.ts]: Thanks, I am stumbling a bit on the api references. Thought I got a "could not find `intgration.scmxyz` ref" error without it, but seems that might have been something else. Removing it to see if all tests still pass

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts
Comment (Rugvip) [reply plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts:43]: Eh, yeah but that's also prolly a bit on and a weird assumption, doesn't hold true for out instance anyway :grin:

I'm thinking remove the check and the tests that expect it, agree with that @backstage/techdocs-core?

Comment (freben) [reply plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts:43]: Agree. Not sure if users will be affected by that though? I don't remember why it was added in the first place, because it sure is a bit of an odd assumption, looking back

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr5461-moipa5ic1r8hftdm58ntizv2uxsibgyp.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/608acd708e7af6514fcc19cd

File: plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts
Comment (freben) [reply plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts:43]: @erikxiv Mind removing the second check? After that this is definitely ready to merge

Comment (erikxiv) [reply plugins/techdocs/src/reader/transformers/addGitFeedbackLink.ts]: Not at all. Pushed.

**Review by freben — APPROVED** [review]

**Pull request merged**
